v2/AdminCore.WebApp/Controllers/HolidayController.cs

```diff
@@ -22,74 +27,184 @@ namespace AdminCore.WebApi.Controllers
   [ApiController]
   public class HolidayController : ControllerBase
   {
+    private readonly IAuthenticatedUser _authenticatedUser;
     private readonly IEventService _holidayEventService;
-
+    private readonly IEmployeeService _employeeService;
     private readonly IMapper _mapper;
 
-    public HolidayController(IEventService holidayEventService, IMapper mapper)
+    public HolidayController(IAuthenticatedUser authenticatedUser, IEventService holidayEventService,
+      IEmployeeService employeeService, IMapper mapper)
     {
+      _authenticatedUser = authenticatedUser;
       _holidayEventService = holidayEventService;
+      _employeeService = employeeService;
       _mapper = mapper;
     }
 
     [HttpGet]
     public IActionResult GetAllHolidays()
     {
-      return Ok();
+      var holidaysDtos = _holidayEventService.GetByType(EventTypes.AnnualLeave);
+      var holidays = _mapper.Map<ResponseMessage<IList<EventDto>>, List<HolidayViewModel>>(holidaysDtos);
+      return Ok(holidays);
     }
 
     [HttpGet("{holidayId}")]
     public IActionResult GetHolidayByHolidayId(int holidayId)
     {
-      return Ok();
+      var holiday = _holidayEventService.GetEvent(holidayId);
+      if (holiday.Status == MessageConstants.MsgStatusSuccess)
+      {
+        return Ok(_mapper.Map<HolidayViewModel>(holiday.Payload));
```

Comment:
There's now inconsistency between how you are doing your returns. Niall is using the base method and Jamie is using this. Can you agree on one approach and both use it please.